reads min/target sdk versions from localproperties

brunotacca opened the pull request:
This PR changes the `build.gradle` files to read the `flutter.minSdkVersion` and `flutter.targetSdkVersion` from the generated local.properties, and if not found, keep the default one (which is set at flutter.gradle)

It changes all `build.gradle` files from examples/* and the integration_test/example.
It doesn't change `build.gradle` files from dev projects at dev/* but they can be changed too upon further request.
It changes both java and kotlin `build.gradle` templates.
It changes the default app test to expect the new strings on `build.gradle`.
It doesn't apply the same concept to `flutter.compileSdkVersion`, so having it on `local.properties` doesn't change the value set at `flutter.gradle`, this can also be added upon further request.

This PR also changes `flutter.gradle` to detect and warn the developer with a `logger.quiet` message about mismatches with plugins API level. If a mismatch is found, it also checks if the variables `flutter.minSdkVersion` and `flutter.targetSdkVersion` were set at `local.properties`

It also changes the `gradle_errors.dart` to update the box message about how to fix the API level error, while adding a reference to the docs. The test was also updated.

**Images:**

<details>
<summary>[Before] New project with a dependency requiring higher sdk version.</summary>

![image](https://user-images.githubusercontent.com/6067887/153695824-305385c1-026d-4b49-97c0-b5c5a9ca4c9c.png)

</details>

<details>
<summary>[After] New project with a dependency requiring higher sdk version.</summary>

![image](https://user-images.githubusercontent.com/6067887/153695696-060bf1f2-6100-4f17-95be-4ef56d878e4e.png)

</details>

<details>
<summary>[After] New project with the variables INcorrectly set at local.properties.</summary>

![image](https://user-images.githubusercontent.com/6067887/153695961-2f3dfbce-44ea-4188-8e76-e322920f4d5d.png)

</details>

<details>
<summary>[After] New project with the variables correctly set at local.properties.</summary>

![image](https://user-images.githubusercontent.com/6067887/153695920-fd6a7dac-90af-4611-b5b0-8a7a31d7298d.png)

</details>

It closes #95533
Partially solves #95485 (Android part)
Could be a solution to #95514

**Extra - Not related:** at https://github.com/flutter/flutter/pull/95684/commits/044079046a3765cfba82991e7002a3339af9b7d7 I fixed the command required to export the `FLUTTER_ROOT` variable in order to run tests.

brunotacca committed:
reads min/target sdk versions from localproperties

meysammahfouzi commented:
How are the values of `flutter.minSdkVersion` and `flutter.targetSdkVersion` initialized at the moment?

brunotacca commented:
Hello Meysam, iirc they are hard coded at flutter.gradle file.

On Sun, Dec 26, 2021, 08:35 Meysam ***@***.***> wrote:

> How are the values of flutter.minSdkVersion and flutter.targetSdkVersion
> initialized at the moment?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flutter/flutter/pull/95684#issuecomment-1001160909>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABOJNL64V5RRRTZFC43EX33US34X7ANCNFSM5KS2TGSQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

christopherfujino commented:
@brunotacca this has merge conflicts with upstream, can you rebase or merge in master to resolve before this is reviewed?

fbcouch committed:
Re-land "Support Scribble Handwriting" (#96615) (#96881)

engine-flutter-autoroll committed:
Roll Engine from 83e3a1c1d55d to 5fa79de88daf (1 revision) (#97400)

brunotacca commented:
> @brunotacca this has merge conflicts with upstream, can you rebase or merge in master to resolve before this is reviewed?

Thank you for notifying me.
There was a conflict with 0e2f51dfd04eca2935318389a75e8bc112809342 on `packages/flutter_tools/test/commands.shard/permeable/create_test.dart` since it changed the same test to verify the build.gradle contents, the `ndkVersion flutter.ndkVersion` was added.

justinmc committed:
Revert "Re-land "Support Scribble Handwriting" (#96615) (#96881)" (#97405)

This reverts commit c54ce01575bf5002529e733143c948ba6ae301b7.

Piinks committed:
Update RawScrollbar to support the track (#97335)

royarg02 committed:
Deprecate `useDeleteButtonTooltip` for Chips (#96174)

engine-flutter-autoroll committed:
Roll Engine from 5fa79de88daf to c5ee2e4c35c6 (3 revisions) (#97417)

TahaTesser committed:
`RefreshIndicator`: Add an interactive example (#97254)

TahaTesser committed:
Add `CupertinoTimerPicker` Interactive Example (#93621)

TahaTesser committed:
Fix tappable area for `DropdownButtonFormField` & add `InkWell` to `DropdownButton` (#95906)

engine-flutter-autoroll committed:
Roll Plugins from bd7dfb78d8f9 to 60982c335b75 (10 revisions) (#97435)

chunhtai committed:
fixes navigator to be able to handle route with duplicate page key in… (#97394)

engine-flutter-autoroll committed:
Roll Engine from c5ee2e4c35c6 to 46bea73d2e04 (1 revision) (#97438)

engine-flutter-autoroll committed:
Roll Engine from 46bea73d2e04 to c2b6e134831a (6 revisions) (#97445)

TahaTesser committed:
Update `PopupMenuButton` example (#96681)

engine-flutter-autoroll committed:
Roll Engine from c2b6e134831a to 2d3663cc4dcf (3 revisions) (#97450)

christopherfujino committed:
[flutter_tools] remove dummy implementation from abstract class .isEnabled() (#96888)

keyonghan committed:
Mark `Windows tool_integration_tests_1_6` flaky (#97458)

engine-flutter-autoroll committed:
Roll Engine from 2d3663cc4dcf to 4b554923f27f (4 revisions) (#97463)

* 7b26f7a16 Roll Skia from 32f7fa7bfef3 to c4ba48a7d8c3 (2 revisions) (flutter/engine#31139)

* 312551e45 Revert "Revert engine (#31135)" (flutter/engine#31142)

* e17ee4a5f Add constexpr endianness utilities to fml (flutter/engine#31118)

* 4b554923f Remove unused Counter and CounterValues class. (flutter/engine#31130)

godofredoc committed:
Set lock github action workflow permissions explicitly. (#97459)

It is a good practice to set the github actions default permissions
to the minimum required. The locks action didn't set permissions
explicitly.

engine-flutter-autoroll committed:
Roll Engine from 4b554923f27f to b74520b22e77 (2 revisions) (#97465)

engine-flutter-autoroll committed:
Roll Engine from b74520b22e77 to fccb9b670d13 (1 revision) (#97474)

DanTup committed:
Make DAP tests more tolerant of output that didn't come from the app being tested (#97291)

Fixes #97238.

guidezpl committed:
Add macOS ephemeral to gitignore (#96397)

guidezpl committed:
[Icons] Prevent double underscores after replacement (#96904)

* Prevent double underscores after replacement

* add test

* remove trailing space

werainkhatri committed:
improve docs for testing dart fix (#97493)

engine-flutter-autoroll committed:
Roll Engine from fccb9b670d13 to b8784efc7ea8 (1 revision) (#97508)

engine-flutter-autoroll committed:
Roll Engine from b8784efc7ea8 to c6e12eb12bb2 (2 revisions) (#97510)

engine-flutter-autoroll committed:
Roll Engine from c6e12eb12bb2 to c2bcffb07e12 (1 revision) (#97512)

engine-flutter-autoroll committed:
Roll Plugins from 60982c335b75 to 72dff1b43426 (1 revision) (#97523)

caioagiani committed:
packages: fix typos recieve => receive (#97488)

engine-flutter-autoroll committed:
Roll Engine from c2bcffb07e12 to 55762740ff96 (2 revisions) (#97527)

engine-flutter-autoroll committed:
Roll Engine from 55762740ff96 to 48ae905cb189 (2 revisions) (#97533)

* f161d762d Roll Skia from 994c946bd33e to 1fad4c0080f2 (2 revisions) (flutter/engine#31160)

* 48ae905cb Export benchmark draw op count in benchmark output and some minor fixes/tweaks (flutter/engine#31146)

engine-flutter-autoroll committed:
Roll Engine from 48ae905cb189 to 800afbf27ac8 (4 revisions) (#97538)

engine-flutter-autoroll committed:
Roll Engine from 800afbf27ac8 to cf26797318ae (2 revisions) (#97548)

engine-flutter-autoroll committed:
Roll Engine from cf26797318ae to 5005e2355eb4 (5 revisions) (#97554)

godofredoc committed:
Pin dependencies in docker file. (#97466)

* Pin dependencies in docker file.

The docker file had two dependencies that were not locked. An npm
dependency and the debian base image. This PR pins those two
dependencies.

* Use the slim version of debian image.

* Use image with bigger disk to build docker images.

* remove web_engine_tests from cirrus.

moffatman committed:
PointerDeviceKind and ui.PointerChange forwards-compatibility (#97350)

* PointerDeviceKind and ui.PointerChange forwards-compatibility

* Add link to trackpad gestures issue

* Add TODO

engine-flutter-autoroll committed:
Roll Engine from 5005e2355eb4 to f15cfb2ccc14 (1 revision) (#97555)

engine-flutter-autoroll committed:
Roll Engine from f15cfb2ccc14 to 740d1420c89a (1 revision) (#97560)

engine-flutter-autoroll committed:
Roll Engine from 740d1420c89a to 7ac2502fe52e (1 revision) (#97563)

engine-flutter-autoroll committed:
Roll Engine from 7ac2502fe52e to fc8895bd4704 (1 revision) (#97565)

engine-flutter-autoroll committed:
Roll Engine from fc8895bd4704 to d653a9e2dd81 (1 revision) (#97568)

engine-flutter-autoroll committed:
Roll Engine from d653a9e2dd81 to 9cd90e399e23 (2 revisions) (#97570)

fbcouch committed:
Re-land "Support Scribble Handwriting" (#96615) (#97437)

christopherfujino committed:
increase timeout for integration tests (#97552)

engine-flutter-autoroll committed:
Roll Engine from 9cd90e399e23 to b0e6fe8e8006 (2 revisions) (#97585)

engine-flutter-autoroll committed:
Roll Plugins from 72dff1b43426 to a5b97ea9b372 (1 revision) (#97590)

engine-flutter-autoroll committed:
Roll Engine from b0e6fe8e8006 to 20de5d73c690 (4 revisions) (#97594)

godofredoc committed:
Do not run score cards on forks of flutter/flutter. (#97603)

As a general rule all the .github workflows enabled for flutter/flutter
should not run on forks.

Bug: https://github.com/flutter/flutter/issues/97599

engine-flutter-autoroll committed:
Roll Engine from 20de5d73c690 to 802332a52524 (5 revisions) (#97601)

jmagman committed:
Disallow running on unsupported devices (#97338)

jmagman committed:
Export an IPA for distribution via "flutter build ipa" without --export-options-plist (#97243)

TahaTesser committed:
BottomNavigationBar: add themeable mouse cursor (#96736)

engine-flutter-autoroll committed:
Roll Engine from 802332a52524 to 9018a102257d (1 revision) (#97605)

jmagman committed:
Run devicelab test xcarchive export as development method (#97615)

jmagman committed:
Revert "Export an IPA for distribution via "flutter build ipa" without --export-options-plist" (#97616)

engine-flutter-autoroll committed:
Roll Engine from 9018a102257d to f22420acf7e1 (2 revisions) (#97610)

engine-flutter-autoroll committed:
Roll Engine from f22420acf7e1 to 7c24d62a4315 (2 revisions) (#97617)

* d0f2134e1 Roll Skia from 074a016b89ef to bc1e97ee5f64 (4 revisions) (flutter/engine#31196)

* 7c24d62a4 Roll Dart SDK from 64532ff7b812 to d2484ba0b7b3 (5 revisions) (flutter/engine#31197)

engine-flutter-autoroll committed:
5bbdb7a6f Do not run score cards in forks of engine. (flutter/engine#31199) (#97620)

engine-flutter-autoroll committed:
Roll Engine from 5bbdb7a6f730 to 0bd729b97242 (1 revision) (#97626)

guidezpl committed:
Update packages (#97592)

* update packages

* implement

* remove trailing spaces

* Revert "remove trailing spaces"

This reverts commit d3bffbef57d142a4a5d98dea4c3e9b6db856c9bb.

* Revert "implement"

This reverts commit 6b6172a827c84985c5fe09e676096878fba43306.

engine-flutter-autoroll committed:
Roll Engine from 0bd729b97242 to e3877ce4fcdd (4 revisions) (#97637)

engine-flutter-autoroll committed:
Roll Engine from e3877ce4fcdd to 4ff105d6256f (2 revisions) (#97642)

engine-flutter-autoroll committed:
Roll Engine from 4ff105d6256f to 2e25b55cc2a4 (2 revisions) (#97645)

collinjackson committed:
[flutter_tools] Use proper project name in templates (#96373)

zanderso committed:
Include Dart SDK arch in package info (#97278)

WasserEsser committed:
Fix VisualDensity's vertical lerp implementation (#96597)

Fixes #96591

albertodev01 committed:
chore: Updated AutofillContextAction documentation (#97245)

zanderso committed:
[flutter_tool] Download gen_snapshot.zip for macOS desktop (#97627)

darrenaustin committed:
Updated gen_defaults to use new JSON output from the Material token database. (#97596)

chingjun committed:
Change all instance of throwing strings to throw specific error classes. (#97325)

zanderso committed:
Revert "[flutter_tool] Download gen_snapshot.zip for macOS desktop (#97627)" (#97664)

This reverts commit 32abf42a0074776542bbbfcb968e25269dee7ddd.

Piinks committed:
Allow Clip.none as a valid clipBehavior (#95593)

chingjun committed:
Addresses the feedbacks in #95738 (#97457)

gaaclarke committed:
Added performance benchmark for background platform channels (#92130)

dcharkes committed:
Migrate bots/test.dart from `pub run` to `dart run` (#97575)

zanderso committed:
Reland: [flutter_tool] Download gen_snapshot.zip for macOS desktop (#97671)

jason-simmons committed:
Exit the tool if a DevTools subprocess fails when running on a bot (#97613)

andreidiaconu committed:
Add DisplayFeatureSubScreen widget (#92907)

Commit:
[flutter_tool] allow disabling profile mode timeline traces (#97622)

darrenaustin committed:
Updated the defaults to only use function overrides for computed values. (#97677)

engine-flutter-autoroll committed:
Roll Plugins from a5b97ea9b372 to 4450606ac080 (1 revision) (#97654)

engine-flutter-autoroll committed:
Roll Engine from 2e25b55cc2a4 to 2c6464e3034f (2 revisions) (#97655)

keyonghan committed:
Revert "Mark `Windows tool_integration_tests_1_6` flaky" (#97665)

fluttergithubbot committed:
Marks Linux_android opacity_peephole_grid_of_alpha_savelayers_perf__e2e_summary to be unflaky (#97667)

fluttergithubbot committed:
Marks Linux_android opacity_peephole_col_of_alpha_savelayer_rows_perf__e2e_summary to be unflaky (#97668)

Piinks committed:
Add docs about horizontal scrollbars (#97673)

Piinks committed:
Update SliverChildDelegate docs (#97674)

engine-flutter-autoroll committed:
Roll Engine from 2c6464e3034f to baadc162a987 (14 revisions) (#97688)

jmagman committed:
Export an IPA for distribution via "flutter build ipa" without --export-options-plist (#97672)

nayeemtby committed:
Add splashBorderRadius to TabBar (#97204)

jason-simmons committed:
Invalidate the TextPainter line metrics cache when redoing text layout (#97446)

engine-flutter-autoroll committed:
Roll Plugins from 4450606ac080 to dbbd21fcd6b1 (3 revisions) (#97708)

dcharkes committed:
Pin `package:ffi` in `plugin_ffi` template (#97720)

engine-flutter-autoroll committed:
Roll Engine from baadc162a987 to e5433ffef929 (2 revisions) (#97718)

fluttergithubbot committed:
Marks Linux_android opacity_peephole_grid_of_opacity_perf__e2e_summary to be unflaky (#97140)

chunhtai committed:
Fix RouterObserver didPop is not called when reverseTransitionDuratio… (#97171)

engine-flutter-autoroll committed:
Roll Engine from e5433ffef929 to 20d779ff7c17 (6 revisions) (#97726)

TahaTesser committed:
Fix Local `SwitchTheme` not being inherited by `Switch` Widget (#97705)

hterkelsen committed:
Add a benchmark for a scrolling list of Links (#94638)

Hixie committed:
Clean up the bindings APIs. (#89451)

TahaTesser committed:
Fix Local `CheckBoxTheme` not being inherited by `CheckBox` Widget (#97715)

engine-flutter-autoroll committed:
Roll Engine from 20d779ff7c17 to 712512d94ee9 (4 revisions) (#97736)

godofredoc committed:
Add dependabot for docker files. (#97556)

engine-flutter-autoroll committed:
Roll Engine from 712512d94ee9 to 82aad2e48051 (2 revisions) (#97743)

engine-flutter-autoroll committed:
Roll Engine from 82aad2e48051 to 98340f9b01d3 (4 revisions) (#97751)

engine-flutter-autoroll committed:
Roll Engine from 98340f9b01d3 to dfaad2a2c2f0 (3 revisions) (#97757)

stuartmorgan-g committed:
Restructure Windows plugin template (#93511)

TahaTesser committed:
Fix Local `RadioTheme` not being inherited by `Radio` Widget (#97713)

engine-flutter-autoroll committed:
Roll Engine from dfaad2a2c2f0 to 0ceed87a2dc2 (4 revisions) (#97777)

engine-flutter-autoroll committed:
Roll Engine from 0ceed87a2dc2 to 0d39b6dd2853 (1 revision) (#97784)

engine-flutter-autoroll committed:
Roll Engine from 0d39b6dd2853 to dbe4cc89ba33 (1 revision) (#97787)

engine-flutter-autoroll committed:
Roll Engine from dbe4cc89ba33 to 0ebbea1766ef (1 revision) (#97791)

engine-flutter-autoroll committed:
Roll Plugins from dbbd21fcd6b1 to c5c8cda753c2 (6 revisions) (#97802)

engine-flutter-autoroll committed:
Roll Engine from 0ebbea1766ef to 2c3ee4895c7f (1 revision) (#97805)

TahaTesser committed:
fix PopupMenuButton example tag (#97763)

engine-flutter-autoroll committed:
Roll Engine from 2c3ee4895c7f to 91a7ee711403 (1 revision) (#97810)

godofredoc committed:
Add dependabot for github actions workflows. (#97760)

engine-flutter-autoroll committed:
Roll Engine from 91a7ee711403 to 1232caf92f83 (2 revisions) (#97817)

fluttergithubbot committed:
Marks Linux_android opacity_peephole_fade_transition_text_perf__e2e_summary to be unflaky (#97144)

* Marks Linux_android opacity_peephole_fade_transition_text_perf__e2e_summary to be unflaky

* trigger builds

Co-authored-by: Keyong Han <keyonghan@google.com>

jpnurmi committed:
docs: adjust the flutter.material.RawMaterialButton.mouseCursor template (#88470)

MichaelTamm committed:
[flutter_tools] Fix bundle file not found when flavor contains upperc… (#92660)

chingjun committed:
Fix error in type cast. (#97778)

devoncarew committed:
use 'dart pub' instead of calling the pub executable directly (#97749)

use 'dart pub' instead of calling the pub executable directly

justinmc committed:
Text editing shift + tap + drag interaction (#95213)

Supports the desktop text editing interaction of holding shift, tapping the field, and dragging to modify the selection.

christopherfujino committed:
[flutter_tools] increment y instead of m when calling flutter --version on master (#97827)

engine-flutter-autoroll committed:
Roll Engine from 1232caf92f83 to f55d4feba892 (4 revisions) (#97834)

jmagman committed:
Include -isysroot -arch and -miphoneos-version-min when creating dummy module App.framework (#97689)

engine-flutter-autoroll committed:
Roll Engine from f55d4feba892 to f04ac2a4fdd0 (2 revisions) (#97841)

engine-flutter-autoroll committed:
Roll Engine from f04ac2a4fdd0 to 89df0d0d87dc (2 revisions) (#97847)

dependabot[bot] committed:
Bump ossf/scorecard-action from 1.0.2 to 1.0.3 (#97819)

dependabot[bot] committed:
Bump github/codeql-action from 1.0.26 to 1.0.31 (#97820)

ColdPaleLight committed:
Make the application lifecycle does not affect `SchedulerBinding.scheduleForcedFrame`. (#97468)

engine-flutter-autoroll committed:
Roll Engine from 89df0d0d87dc to 45556e526582 (2 revisions) (#97850)

engine-flutter-autoroll committed:
Roll Engine from 45556e526582 to 5a091b756ea9 (2 revisions) (#97852)

engine-flutter-autoroll committed:
Roll Engine from 5a091b756ea9 to bbe4e9758c7b (1 revision) (#97854)

cachapa committed:
Report progress on Dismissible update callback (#95504)

engine-flutter-autoroll committed:
Roll Engine from bbe4e9758c7b to ea8386542aef (4 revisions) (#97881)

engine-flutter-autoroll committed:
Roll Engine from ea8386542aef to 2a4709ae9fc4 (3 revisions) (#97890)

engine-flutter-autoroll committed:
Roll Engine from 2a4709ae9fc4 to 0712096899f0 (1 revision) (#97892)

DanTup committed:
Add support for attachRequest in DAP, running "flutter attach" (#97652)

* Add support for attachRequest in DAP, which runs "flutter attach"

* Update DAP docs for attachRequest

* Improve doc comments

* Fix comments

* Remove noDebug from attach + create a getter for `debug`

* Fix indent

Commit:
Fix how Gradle resolves Android plugin (#97823)

Commit:
Pin Gallery to version with updated settings.gradle (#97986)

WasserEsser committed:
RenderIndexedStack - Mark invisible children as offstage in debugDescribeProperties (#96639)

TahaTesser committed:
TabBar: add themeable mouse cursor  (#96737)

werainkhatri committed:
removes Material (#96899)

darrenaustin committed:
Material 3 Typography support. (#97829)

justinmc committed:
Undo/redo (#96968)

fluttergithubbot committed:
Marks Mac_ios backdrop_filter_perf_ios__timeline_summary to be flaky (#97959)

fluttergithubbot committed:
Marks Mac_ios simple_animation_perf_ios to be flaky (#97962)

chunhtai committed:
Remove RenderEditable dependency from TextSelectionHandleOverlay (#97967)

engine-flutter-autoroll committed:
Roll Engine from 0712096899f0 to bfb58101c185 (2 revisions) (#97981)

dependabot[bot] committed:
Bump github/codeql-action from 1.0.31 to 1.0.32 (#97983)

Commit:
[ci] Remove ios32 tests (#97385)

LongCatIsLooong committed:
Use ValueKey finder instead of text finder in keyboard_textfield_test (#97996)

keyonghan committed:
Marks Mac module_test_ios to be flaky (#97997)

engine-flutter-autoroll committed:
Roll Engine from bfb58101c185 to 1c989109e39f (7 revisions) (#98000)

Commit:
Add a benchmark for simple DisplayLists to exercise the Raster Cache's cache admission algorithm (#97992)

engine-flutter-autoroll committed:
Roll Engine from 1c989109e39f to e886029bf4f8 (2 revisions) (#98002)

dependabot[bot] committed:
Bump dessant/lock-threads from 2.0.3 to 3 (#97821)

Commit:
[framework] don't hit test for system nav bar or system chrome on desktop (#97883)

godofredoc committed:
Remove version comments from workflow. (#98001)

engine-flutter-autoroll committed:
Roll Engine from e886029bf4f8 to 8c3549974d4a (1 revision) (#98005)

engine-flutter-autoroll committed:
Roll Engine from 8c3549974d4a to 2ba0b934f4b7 (1 revision) (#98010)

engine-flutter-autoroll committed:
Roll Engine from 2ba0b934f4b7 to c797abff336f (3 revisions) (#98019)

engine-flutter-autoroll committed:
Roll Engine from c797abff336f to 7bf74069a5cd (1 revision) (#98025)

engine-flutter-autoroll committed:
Roll Plugins from c5c8cda753c2 to 99feee9996cb (15 revisions) (#98037)

fluttergithubbot committed:
Marks Windows gradle_plugin_light_apk_test to be flaky (#97957)

bannzai committed:
Fix typo FutterApplication -> FlutterApplication (#97897)

Commit:
Revert "Fix how Gradle resolves Android plugin" (#98050)

Commit:
[framework] inline casts on Element.widget getter to improve web performance (#97822)

Commit:
Split complexity scoring tests into two tasks - one for Android and one for iOS (#98063)

godofredoc committed:
Remove securitycards scheduled runs. (#98062)

LongCatIsLooong committed:
[EditableText] honor the "brieflyShowPassword" system setting (#97769)

godofredoc committed:
Revert "[EditableText] honor the "brieflyShowPassword" system setting (#97769)" (#98089)

This reverts commit 63f48d1ff8dbecc6adbc8fd622e93a493dadbc26.

gslender committed:
implemented TapUp within InkResponse and InkWell (#93833)

* implemented TapUp within InkResponse and InkWell

* Update packages/flutter/lib/src/material/ink_well.dart

Co-authored-by: Tong Mu <dkwingsmt@users.noreply.github.com>

* Update packages/flutter/lib/src/material/ink_well.dart

Co-authored-by: Viren Khatri <werainkhatri.work@gmail.com>

* Update packages/flutter/lib/src/material/ink_well.dart

Co-authored-by: Viren Khatri <werainkhatri.work@gmail.com>

Co-authored-by: Tong Mu <dkwingsmt@users.noreply.github.com>
Co-authored-by: Viren Khatri <werainkhatri.work@gmail.com>

Commit:
Export activity for module in AndroidManifest.xml (#98076)

engine-flutter-autoroll committed:
Roll Engine from 7bf74069a5cd to 2eaed3c7c3ce (10 revisions) (#98078)

engine-flutter-autoroll committed:
Roll Engine from 2eaed3c7c3ce to 3a381be8dcdd (6 revisions) (#98100)

wangying3426 committed:
Fix timer keeps active when resampling disabled in some cases (#97197)

engine-flutter-autoroll committed:
Roll Engine from 3a381be8dcdd to f3ebba540bf6 (2 revisions) (#98106)

guidezpl committed:
Update repo generated_plugins.cmake (#98034)

engine-flutter-autoroll committed:
Roll Plugins from 99feee9996cb to 9d32c83af9af (1 revision) (#98127)

engine-flutter-autoroll committed:
Roll Engine from f3ebba540bf6 to 3dd42fd7f1b1 (1 revision) (#98129)

chunhtai committed:
Unify Text selection API (#98073)

* Unify Text selection API

* fix test

* further simplify

* fix more tests

* styling change

dcharkes committed:
Add debugging info to android_plugin_example_app_build_test (#98107)

engine-flutter-autoroll committed:
Roll Engine from 3dd42fd7f1b1 to a57061eda8a1 (1 revision) (#98135)

chunhtai committed:
Allow remove listener on disposed change notifier (#97988)

Commit:
[flutter_driver] show refresh rate status in timeline summary (#95699)

jmagman committed:
Override OS and Xcode version for plugin_lint_mac (#98210)

jmagman committed:
Revert "Update plugin lint test for federated url_launcher plugin (#94374)" (#98166)

This reverts commit 86b80d1cfeae345e3bdabebc8491fbea39ed4e97.

justinmc committed:
Shift tap on an unfocused field (#97543)

jmagman committed:
Pin url_launcher to federated version in plugin_lint_mac (#98171)

justinmc committed:
Windows/Linux keyboard shortcuts at a wordwrap (#96323)

chunhtai committed:
Relax android semantics integration test checks (#97742)

moffatman committed:
PointerDeviceKind forwards-compatibility in flutter_test (#98202)

LongCatIsLooong committed:
EditableText does not request focus on autofill (#97846)

engine-flutter-autoroll committed:
bed591d39 Roll Skia from 3e6e29cf2e1b to a424b619bc40 (3 revisions) (flutter/engine#31349) (#98142)

jonahwilliams committed:
[framework] use platform touchslop on Android (#97971)

TahaTesser committed:
Fix NavigationRail Indicator alignment for `NavigationRailLabelType.none` (#98028)

darrenaustin committed:
Update Material tokens to v0.81. (#98149)

brianquinlan committed:
Add `keyLog` and `connectionFactory` to HttpClient implementations (#98045)

swift-kim committed:
[flutter_tools] Make variants of Pub have consistent method signatures (#98119)

yusuf-goog committed:
Update Xcode version to 13a233 (#98144)

TahaTesser committed:
Fix flutter gen-l10n help message (#98147)

LongCatIsLooong committed:
Reland "[EditableText] honor the "brieflyShowPassword" system setting #97769 " (#98150)

TahaTesser committed:
[integration_test] Fix official example for Android (#98162)

goderbauer committed:
[performance] Process dirty nodes from top to bottom during paint to avoid unnecessary layer tree walks (#98219)

zanderso committed:
Revert "Update Xcode version to 13a233 (#98144)" (#98263)

This reverts commit 84fcb158e85735da99423f36ac8a9f9285a16b65.

camsim99 commented on packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl:
```diff
@@ -25,6 +25,18 @@ apply plugin: 'com.android.application'
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+    logger.quiet("flutter.minSdkVersion not found on android/local.properties, consider adding it. Using ${flutterMinSdkVersion} as default.")
```
Is it something to warn about if the developer doesn't specify it in local.properties? I suppose adding this warning could help with [Issue #95514](https://github.com/flutter/flutter/issues/95514), though. Is that your reasoning?

Same comment for `flutterTargetSdkVersion`.

brunotacca replied:
Exactly that, it could prevent similar issues in the future. Although, since now there is mention of the `local.properties` on the flutter website docs, I don't know if it's that necessary anymore.

camsim99 replied:
Oh gotcha! Could we possibly move the warning to [`flutter.gradle`](https://github.com/flutter/flutter/blob/stable/packages/flutter_tools/gradle/flutter.gradle) and only show it when not defining it will be a problem, i.e. when the `minSdkVersion` of a dependency is higher than that of the flutter project?

brunotacca replied:
Certainly, I gonna need a little bit of time to understand how `flutter.gradle` works but I am on it. Let me get this straight, ideally, what would be the conditions to trigger the custom warning about `local.properties`?

camsim99 replied:
It seems like if the minSdkVersion of a dependency is greater than that of the mindSdkVersion defined in `flutter.gradle`, we show the warning so that the person knows how to fix the error that will appear. You can even add a link to your update on the website to the warning for context!

It should be similar to the way we compare the `compileSdkVersion`s of the project and its plugin dependencies right now (see [here](https://github.com/flutter/flutter/blob/61ac285608cd3c3c003507a572b34fad893a21d4/packages/flutter_tools/gradle/flutter.gradle#L450)), but let me know if I can help!

brunotacca replied:
Thank you so much, that should be more than enough for me to get started. I will let you know if I get stuck, thanks!!

brunotacca replied:
@camsim99 Updated the PR description, images. I changed `flutter.gradle` and also `gradle_errors.dart`. 

Thank you for reviewing.

KristinBi committed:
Add benchmark to property (#98159)

keyonghan committed:
Support Samsung phone when getting device `wakefulness` status (#98146)

chunhtai committed:
Refactor TextSelectionOverlay (#98153)

* Refactor TextSelectionOverlay

* fix test

* remove print

* fix more tests

* update

* added tests

* fix comments

* fmt

* fix test

* addressing comment

* remove dispose

* remove new line

goderbauer committed:
[performance] Trace direct calls to inflateWidget (#98277)

brunotacca committed:
logger message removed from templates

brunotacca committed:
plugin minSdk check with help msg.

brunotacca committed:
fix gradle_errors to use minSdk local.properties

brunotacca committed:
fixing command on readme

camsim99 reviewed (approved):
LGTM! Just need to get the tests passing :)

christopherfujino commented:
To get customer_testing to pass, you will need to rebase (not merge) upstream master.

brunotacca commented:
@christopherfujino It seems that I did something wrong trying to rebase... Should I close this and make another PR?

christopherfujino commented:
> @christopherfujino It seems that I did something wrong trying to rebase... Should I close this and make another PR?

That works if you are not able to resolve the bad rebase.

brunotacca commented:
@christopherfujino #98450 I think I did it correctly this time, sorry for the inconvenience.
